### PR TITLE
Refactoring viewer

### PR DESCRIPTION
**Pull request by thorvaldj**

### CODE REVIEW

File: applications/segyviewer.py [hunk 1]
```diff
@@ -1,410 +1,146 @@
...
+        def perform_colormap_change():
+            colormap_monitor.colormap_updated(color_map_name)
 
-    def set_index(self, val):
-        self.sbox.blockSignals(True)
-        self.sbox.setValue(val)
-        self.sbox.blockSignals(False)
+        return perform_colormap_change
 
+    for item in ['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']:
```
Comment (pgdr) [inline applications/segyviewer.py]: Perhaps extract `['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']` to a variable at class level?  E.g.
```python
available_colormaps = ['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']
```

File: python/segyview/__init__.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from .segyplot import SegyPlot
+import util
```
Comment (pgdr) [inline python/segyview/__init__.py]: should you use `import .util` in this case?

File: tests/test_segy.py [hunk 3]
```diff
@@ -111,6 +111,90 @@ def test_xline_22(self):
...
+                        self.assertEqual(self.il_sample(sample), ilno,
+                                         ("sample: {0}, ilno {1}".format(self.il_sample(sample), ilno)))
+                        self.assertEqual(self.xl_sample(sample), xlno,
+                                         ("sample: {0}, xlno {1}, sample {2}".format(
+                                             self.xl_sample(sample), xlno, sample)))
+                        self.assertEqual(self.depth_sample(sample), sample_index,
+                                         ("sample: {0}, sample_index {1}, real_sample {2}".format(
+                                             self.depth_sample(sample), sample_index, sample)))
+
+    def test_read_all_depth_slices(self):
```
Comment (jepebe) [inline tests/test_segy.py]: These tests and the `make_file` function should be removed.

File: tests/test_segyview.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+from unittest import TestCase
+import segyio
+import segyview.util as util
+import itertools
+from test_segy import TestSegy
+
+
+class TestSegyView(TestCase):
```
Comment (jepebe) [inline tests/test_segyview.py]: These tests should just read the `test-data/small.sgy` and use the `SegyIO API` to compare between memory and file.

File: applications/segyviewer.py [hunk 5]
```diff
@@ -1,410 +1,147 @@
...
+import sys
 import segyio
-
-from pylab import *
 from PyQt4 import QtGui, QtCore
-from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
-from matplotlib.figure import Figure
-import matplotlib.patches as patches
+from segyview import *
+from segyview import util
```
Comment (kjellkongsvik) [inline applications/segyviewer.py]: you already have import * above. You don't need both.

**Pull request merged**